tests/php/Emitter/OutputEmitter/NodeEmitter/ApplyEmitterTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhelTest\Emitter\OutputEmitter\NodeEmitter;
+
+use Phel\Ast\ApplyNode;
+use Phel\Ast\FnNode;
+use Phel\Ast\LiteralNode;
+use Phel\Ast\PhpVarNode;
+use Phel\Ast\TupleNode;
+use Phel\Emitter\OutputEmitter;
+use Phel\Emitter\OutputEmitter\NodeEmitter\ApplyEmitter;
+use Phel\Lang\Symbol;
+use Phel\NodeEnvironment;
+use PHPUnit\Framework\TestCase;
+
+final class ApplyEmitterTest extends TestCase
+{
+    private ApplyEmitter $applyEmitter;
+
+    public function setUp(): void
+    {
+        $this->applyEmitter = new ApplyEmitter(
+            OutputEmitter::createWithSourceMap()
+        );
+    }
+
+    public function testPhpVarNodeAndFnNodeIsInfix(): void
+    {
+        $node = new PhpVarNode(NodeEnvironment::empty(), '+');
+        $args = [
+            new TupleNode(NodeEnvironment::empty(), [
+                new LiteralNode(NodeEnvironment::empty(), 2),
+                new LiteralNode(NodeEnvironment::empty(), 3),
+                new LiteralNode(NodeEnvironment::empty(), 4),
+            ]),
+        ];
+
+        $applyNode = new ApplyNode(NodeEnvironment::empty(), $node, $args);
+        $this->applyEmitter->emit($applyNode);
+
+        $this->expectOutputString(
+            'array_reduce([...((\Phel\Lang\Tuple::createBracket(, , );) ?? [])], function($a, $b) { return ($a + $b); });'
+        );
+    }
+
+    public function testPhpVarNodeButNoInfix(): void
+    {
+        $node = new PhpVarNode(NodeEnvironment::empty(), 'str');
+        $args = [
+            new LiteralNode(NodeEnvironment::empty(), 'abc'),
+            new TupleNode(NodeEnvironment::empty(), [
+                new LiteralNode(NodeEnvironment::empty(), 'def'),
+            ]),
+        ];
+
+        $applyNode = new ApplyNode(NodeEnvironment::empty(), $node, $args);
+        $this->applyEmitter->emit($applyNode);
+
+        $this->expectOutputString('str(, ...((\Phel\Lang\Tuple::createBracket();) ?? []));');
+    }
+
+    public function testNoPhpVarNode(): void
+    {
+        $fnNode = new FnNode(
+            NodeEnvironment::empty(),
+            [Symbol::create('x')],
+            new PhpVarNode(NodeEnvironment::empty(), 'x'),
+            [],
+            true,
+            false
```

Comment:
```suggestion
        $isVariadic = true,
        $recurs = false
```
Is a way of simulate "named arguments". This way it will be clear what these bool parameters represent. In general, bool arguments are a bad design (because it's readability and temporal-mapping) but a way to improve them is to add a variable name that represents its value.

Reply:
You are right!
Thank you :dancers: